circuit_maintenance_parser/processor.py

```diff
@@ -0,0 +1,119 @@
+                error_message = "Parser class %s from %s was not successful.\n%s"
+                logger.debug(error_message, data_parser.__name__, self.__class__.__name__, traceback.format_exc())
+                raise ProcessorError(error_message % data_parser.__name__, self.__class__.__name__, "") from exc
+
+        self.post_process_hook(maintenances_data)
+
+        return maintenances_data
+
+    def process_hook(self, maintenances_extracted_data: List, maintenances_data: List):
+        """Custom method per processor to accumulate the data from each DataPart."""
+        raise NotImplementedError
+
+    def post_process_hook(self, maintenances_data: List):
+        """Hook to add a post parsing process logic."""
+
+    def extend_processor_data(self, current_maintenance_data):
+        """Method used to extend Maintenance data with some defaults."""
+        current_maintenance_data["organizer"] = (
+            self.extended_data.get("organizer")
+            if "organizer" not in current_maintenance_data
+            else current_maintenance_data.get("organizer")
+        )
+
+        if "provider" not in current_maintenance_data:
+            current_maintenance_data["provider"] = self.extended_data.get("provider")
+
+
+class SimpleProcessor(GenericProcessor):
+    """Processor to get all the Maintenance Data in each Data Part."""
+
+    def process_hook(self, maintenances_extracted_data, maintenances_data):
+        """For each data extracted (that can be multiple), we try to build a complete Maintenance."""
+        for extracted_data in maintenances_extracted_data:
+            self.extend_processor_data(extracted_data)
+            maintenances_data.append(Maintenance(**extracted_data))
+
+
+class CombinedProcessor(GenericProcessor):
+    """Processor that combines the parsed output of multiple DataParts to get a single unified Maintenance notification."""
+
+    # The CombinedProcessor will consolidate all the parsed data into this variable
+    combined_maintenance_data: Dict = {}
+
+    def process_hook(self, maintenances_extracted_data, maintenances_data):
+        """All the parsers contribute with a subset of data that is extended."""
+        # We only expect one data object from these parsers
+        if len(maintenances_extracted_data) == 1:
+            self.combined_maintenance_data.update(maintenances_extracted_data[0])
+        else:
+            raise ProcessorError(f"Unexpected data retrieved from parser: {maintenances_extracted_data}")
+
+    def post_process_hook(self, maintenances_data):
+        """After processing all the parsers, we try to combine all the data together."""
+        self.extend_processor_data(self.combined_maintenance_data)
+        try:
+            maintenances_data.append(Maintenance(**self.combined_maintenance_data))
+        except ValidationError as exc:
+            raise ProviderError("Not enough information available to create a Maintenance notification.") from exc
```

Comment:
I think I suggested `ProviderError` in a previous review comment, but actually, should this be `ProcessorError` now?